### DISCUSSION TITLE
How can I add macro recording information to statusline?

### DISCUSSION THREAD

**Discussion by guillenotfound** [NvChad/NvChad#3125] [Q&A] [answered]
So there's a line under the statusline normally but when using noice that line goes away. I'm wondering if there's a way to display `@<a-z>` or similar in the statusline directly?

I've checked the docs via `:h nvui.statusline` but there's no information regarding macro?

Additionally I'm trying to understand how all pieces are connected, base46 and ui in this case, where can I get more information regarding this two?

In the help I see `- To know which highlight groups are available, check base46 integrations dir` but IDK which is that directory.

I'm pretty sure that there might be some questions that are repeated so maybe by improving a bit the docs it could be beneficial both from a consumer POV and from the maintainer?

Anyways thanks a lot for you job, moved from vscode to neovim (nvchad) more than 1 year ago and I'm very happy! :)

**Comment by siduck** [comment]
how would you do it in normal nvim? statusline is just a list of concaneted lists

read the statusline of nvui properly, make a new module and include it in the order.  make that module return whatever you want

**Comment by vivindeena** [comment, marked as answer]
```lua
---@type ChadrcConfig
local M = {}

M.ui = {
    statusline = {
        order = { "mode", "file", "macro", "git", "%=", "lsp_msg", "%=", "diagnostics", "lsp", "cwd", "cursor" },
        modules = {
            macro = function()
                local reg = vim.fn.reg_recording()
                if reg == "" then
                    return ""
                end
                return "%#WarningMsg# " .. reg .. "%#StText#"
            end,
        },
    },
}
return M
```

hopefully this helps someone in the future

**Reply by siduck** [reply]
@vivindeena  good job!
